AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignOut/ShowHostedUISignOut.swift

```diff
@@ -0,0 +1,128 @@
+        var components = URLComponents()
+        components.scheme = "https"
+        components.path = "/logout"
+        components.host = hostedUIConfig.oauth.domain
+        components.queryItems = [
+            URLQueryItem(name: "client_id", value: hostedUIConfig.clientId),
+            URLQueryItem(name: "logout_uri", value: signOutURI),
+        ]
+
+        guard let logutURL = components.url else {
+            let error = AuthenticationError.configuration(message: "Could not create logout URL")
+            let event = SignOutEvent(eventType: .signedOutFailure(error))
+            logVerbose("\(#fileID) Sending event \(event)", environment: environment)
+            dispatcher.send(event)
+            return
+        }
+
+
+        let aswebAuthenticationSession = ASWebAuthenticationSession(
+            url: logutURL,
+            callbackURLScheme: callbackURLScheme,
+            completionHandler: { url, error in
+
+                if let url = url {
+                    let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false)
+                    let queryItems = urlComponents?.queryItems
+
+                    if let error = queryItems?.first(where: { $0.name == "error" })?.value {
+                        let event = HostedUIEvent(eventType: .throwError(.hostedUI(.serviceMessage(error))))
+                        self.logVerbose("\(#fileID) Sending event \(event)", environment: environment)
+                        dispatcher.send(event)
+                        return
+                    }
+
+                    let event: SignOutEvent
+                    if self.signOutEvent.globalSignOut {
+                        event = SignOutEvent(eventType: .signOutGlobally(self.signInData))
+                    } else {
+                        event = SignOutEvent(eventType: .revokeToken(self.signInData))
+                    }
+                    self.logVerbose("\(#fileID) Sending event \(event.type)", environment: environment)
+                    dispatcher.send(event)
+                }
+
+                if let error = error {
+                    let event = HostedUIEvent(eventType: .throwError(.service(error: error)))
+                    self.logVerbose("\(#fileID) Sending event \(event.type)", environment: environment)
+                    dispatcher.send(event)
+                }
+
+        })
+        aswebAuthenticationSession.presentationContextProvider = self
+        DispatchQueue.main.async {
+            aswebAuthenticationSession.start()
+        }
+    }
+
+}
+
+extension ShowHostedUISignOut: ASWebAuthenticationPresentationContextProviding {
+    func presentationAnchor(for session: ASWebAuthenticationSession) -> ASPresentationAnchor {
+        return ASPresentationAnchor()
+    }
+}
+
+extension ShowHostedUISignOut: CustomDebugDictionaryConvertible {
+    var debugDictionary: [String: Any] {
+        [
+            "identifier": identifier
+        ]
```

Comment:
```
    let signOutEvent: SignOutEventData
    let signInData: SignedInData
```

Add some data points to the debug dictionary.

Reply:
Added